### PR TITLE
add consent to call/text and hasmobile to update patient opts

### PR DESCRIPTION
**Pull request by dillonstreator**
The PUT patient endpoint expects `hasmobileyn` while the GET patient endpoint send `hasmobile`

### CODE REVIEW

File: athenahealth/patients.go [hunk 1]
```diff
@@ -307,6 +309,7 @@ type UpdatePatientOptions struct {
 	FirstName           *string
 	GenderIdentity      *string
 	GenderIdentityOther *string
+	HasMobile           *bool
```
Comment (elliehastings) [inline athenahealth/patients.go:312]: Any reason the client field name doesn't match the API field name? Seems like most others are 1-to-1

File: athenahealth/patients.go [hunk 2]
```diff
@@ -355,6 +358,14 @@ func (h *HTTPClient) UpdatePatient(ctx context.Context, patientID string, opts *
 			form.Add("assignedsexatbirth", *opts.AssignedSexAtBirth)
 		}
 
+		if opts.ConsentToCall != nil {
+			form.Add("consenttocall", strconv.FormatBool(*opts.ConsentToCall))
```
Comment (elliehastings) [inline athenahealth/patients.go:362]: Do these need to be strings even though the API docs describe them as boolean?

File: athenahealth/patients.go (hunk 1 above)
Comment (dillonstreator) [reply athenahealth/patients.go:312]: copied it from the patient record
suppose there's no reason on to just match the API fields directly

File: athenahealth/patients.go (hunk 2 above)
Comment (dillonstreator) [reply athenahealth/patients.go:362]: `url.Values` only accepts string values
this does seem to be working but i think there is an issue with what athena api is documenting.. it seems like a combination of `hasmobile` and `hasmobileyn` is accepted depending on whether getting or updating a patient.

**Review by fterrag — DISMISSED** [review]

**Review by elliehastings — DISMISSED** [review]

File: athenahealth/patients.go (hunk 2 above)
Comment (dillonstreator) [reply athenahealth/patients.go:362]: After testing - i confirmed that the PUT patient endpoint expects `hasmobileyn` while the GET patient endpoint send `hasmobile`

Comment (willfitze) [reply athenahealth/patients.go:362]: 🤘

File: athenahealth/resources/GetPatient.json [hunk 3]
```diff
@@ -107,7 +107,7 @@
         "contactpreference_billing_email": true,
         "mobilephone": "8608105503",
         "contactpreference_announcement_email": true,
-        "hasmobile": true,
+        "hasmobileyn": true,
```
Comment (willfitze) [inline athenahealth/resources/GetPatient.json:110]: Given the conversation above do these GET responses need to be `hasmobile`?

Comment (dillonstreator) [reply athenahealth/resources/GetPatient.json:110]: ah yea that should be updated, although it's not really being used

**Review by willfitze — APPROVED** [review]

**Pull request merged**